src/main/java/com/redhatinsights/relations_connector/RelationsSinkTask.java

```diff
@@ -17,86 +17,157 @@
 import org.apache.kafka.common.TopicPartition;
-import org.apache.kafka.common.config.AbstractConfig;
-import org.apache.kafka.connect.errors.ConnectException;
 import org.apache.kafka.connect.sink.SinkRecord;
 import org.apache.kafka.connect.sink.SinkTask;
+import org.apache.kafka.connect.json.JsonConverter;
+import org.project_kessel.api.relations.v1beta1.*;
+import org.project_kessel.relations.client.RelationTuplesClient;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import static com.redhatinsights.relations_connector.RelationsSinkConnector.startOrRetrieveManagerFromProps;
+
 /**
  * FileStreamSinkTask writes records to stdout or a file.
  */
 public class RelationsSinkTask extends SinkTask {
     private static final Logger log = LoggerFactory.getLogger(RelationsSinkTask.class);
 
-    private String filename;
-    private PrintStream outputStream;
+    private RelationTuplesClient relationTuplesClient;
+    private final JsonConverter jsonConverter = new JsonConverter();
 
     public RelationsSinkTask() {
     }
 
-    // for testing
-    public RelationsSinkTask(PrintStream outputStream) {
-        filename = null;
-        this.outputStream = outputStream;
-    }
-
     @Override
     public String version() {
         return "v0.1";
     }
 
     @Override
     public void start(Map<String, String> props) {
-        AbstractConfig config = new AbstractConfig(RelationsSinkConnector.CONFIG_DEF, props);
-        filename = config.getString(RelationsSinkConnector.FILE_CONFIG);
-        if (filename == null || filename.isEmpty()) {
-            outputStream = System.out;
-        } else {
-            try {
-                outputStream = new PrintStream(
-                    Files.newOutputStream(Paths.get(filename), StandardOpenOption.CREATE, StandardOpenOption.APPEND),
-                    false,
-                    StandardCharsets.UTF_8.name());
-            } catch (IOException e) {
-                throw new ConnectException("Couldn't find or create file '" + filename + "' for FileStreamSinkTask", e);
-            }
-        }
+        log.debug("Starting RelationsSinkTask");
+        jsonConverter.configure(props, false);
+        /* No new grpc channel will be created, only retrieved, since the connector makes the call first */
+        var relationsClientsManager = startOrRetrieveManagerFromProps(props);
+        relationTuplesClient = relationsClientsManager.getRelationTuplesClient();
+        log.trace("Done starting RelationsSinkTask");
     }
 
     @Override
     public void put(Collection<SinkRecord> sinkRecords) {
+        log.trace("Putting sinkRecords");
         for (SinkRecord record : sinkRecords) {
-            log.trace("Writing line to {}: {}", logFilename(), record.value());
-            outputStream.println(record.value());
+            log.trace("Processing record {}", record.value());
+            try {
+                byte[] rawJson = jsonConverter.fromConnectData(
```

Comment:
I'm guessing that in the future the topic is something we'll parameterize? Or just hardcode what we settle on in the end?

Reply:
I think we'll have to parameterize it.

Reply:
Done now.